examples/C++/DDS/ZeroCopyExample/LoanableHelloWorldPublisher.cxx

```diff
@@ -0,0 +1,168 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*!
+ * @file LoanableHelloWorldPublisher.cpp
+ * This file contains the implementation of the publisher functions.
+ *
+ * This file was generated by the tool fastcdrgen.
+ */
+
+
+#include "LoanableHelloWorldPublisher.h"
+#include "LoanableHelloWorldPubSubTypes.h"
+
+#include <fastdds/dds/domain/DomainParticipantFactory.hpp>
+#include <fastdds/dds/publisher/Publisher.hpp>
+#include <fastdds/dds/publisher/qos/PublisherQos.hpp>
+#include <fastdds/dds/publisher/DataWriter.hpp>
+#include <fastdds/dds/publisher/qos/DataWriterQos.hpp>
+
+#include <thread>
+#include <chrono>
+
+using namespace eprosima::fastdds::dds;
+
+LoanableHelloWorldPublisher::LoanableHelloWorldPublisher()
+    : participant_(nullptr)
+    , publisher_(nullptr)
+    , topic_(nullptr)
+    , writer_(nullptr)
+    , type_(new LoanableHelloWorldPubSubType())
+{
+}
+
+LoanableHelloWorldPublisher::~LoanableHelloWorldPublisher()
+{
+    if (writer_ != nullptr)
+    {
+        publisher_->delete_datawriter(writer_);
+    }
+    if (publisher_ != nullptr)
+    {
+        participant_->delete_publisher(publisher_);
+    }
+    if (topic_ != nullptr)
+    {
+        participant_->delete_topic(topic_);
+    }
+    DomainParticipantFactory::get_instance()->delete_participant(participant_);
+}
+
+bool LoanableHelloWorldPublisher::init()
+{
+    /* Initialize data_ here */
+
+    //CREATE THE PARTICIPANT
+    DomainParticipantQos pqos;
+    pqos.name("Participant_pub");
+    participant_ = DomainParticipantFactory::get_instance()->create_participant(0, pqos);
+    if (participant_ == nullptr)
+    {
+        return false;
+    }
+
+    //REGISTER THE TYPE
+    type_.register_type(participant_);
+
+    //CREATE THE PUBLISHER
+    publisher_ = participant_->create_publisher(PUBLISHER_QOS_DEFAULT, nullptr);
+    if (publisher_ == nullptr)
+    {
+        return false;
+    }
+
+    //CREATE THE TOPIC
+    topic_ = participant_->create_topic(
+        "LoanableHelloWorldTopic",
+        type_.get_type_name(),
+        TOPIC_QOS_DEFAULT);
+    if (topic_ == nullptr)
+    {
+        return false;
+    }
+
+    // CREATE THE WRITER
+    DataWriterQos wqos = DATAWRITER_QOS_DEFAULT;
```

Comment:
```suggestion
    DataWriterQos wqos = publisher->get_default_datawriter_qos();
```
`get_default_datawriter_qos` and `DATAWRITER_QOS_DEFAULT` may be different. `get_default_datawriter_qos` applies the default profile, while `DATAWRITER_QOS_DEFAULT` is a default constructed object.
It would not mind on a test, but being an example, I think is more appropriate.

The same applies to the Subscriber.

Reply:
Done in 415db00